### DISCUSSION TITLE
Rate limiting next build of static pages

### DISCUSSION THREAD

**Discussion by ChristineTham** [vercel/next.js#21656] [Ideas]
**Describe the feature you'd like to request**

Is it possible to rate limit next build so that the next build process generates dynamic routes (eg. /posts/[id]) at a max rate (eg. 10 per sec)

**Describe the solution you'd like**

eg. /pages/posts/[id].js

has a GetStaticProps that calls an external API to retrieve data for a page - this API should not be called more than x times per interval.

**Describe alternatives you've considered**

I have tried using bottleneck package to limit the number of times an API is called but it doesn't seem to work because of async/await.

**Comment by JimmyGray** [comment]
+1 this would be super useful!!
